python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -26,11 +26,39 @@
 
 import numpy as np
 
+from rmm import RMMError
+
 from ..options import _env_get_bool
 from ..testing._utils import assert_eq
 from .annotation import nvtx
 
 
+class CudfPandasDebugWarning(UserWarning):
```

Comment:
One line docstrings here would be helpful to clarify when each should be raised

Reply:
A suggestion for the inheritance of these warnings (since these will be just internal for now)

`class CudfPandasWarning(UserWarning)`

`class FallbackWarning(CudfPandasWarning)`

`class OOMWarning(FallbackWarning)`

`class NotImplementedWarning(FallbackWarning)`

etc

Reply:
Could you implement the suggestion above so that we can reduce the verbosity of the warning names?